### PR TITLE
Add logrotate as well as the first stab at CLI

### LINKED ISSUE #17: Add log rotating configuration
We already rotate backups, we should also have easy configuration for logfiles that should get rotated.  Default value should be /u/apps/whatever/production.log

### PR DESCRIPTION
**Pull request by samsonasu**
fixes #17

### CODE REVIEW

**Comment by petitJAM** [comment]
so what would the ansible-vault edit look like in your cli?  https://github.com/tenforwardconsulting/subspace/issues/25

File: lib/subspace/cli.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      c.summary = 'ssh to the remote server as the administrative user'
+      c.description = ''
+      c.when_called Subspace::Commands::Ssh
+    end
+
+    run!
+  end
+end
+
+#MyApplication.new.run if $0 == __FILE__
```
Comment (boatrite) [inline lib/subspace/cli.rb]: :dash: ?

File: lib/subspace/commands/ansible.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  module Commands
+    module Ansible
+      def ansible_command(command, *args)
+        Dir.chdir "config/provision" do
+          say ">> Running #{command} #{args.join(' ')}"
+          system(command, *args, out: $stdout, err: $stderr)
+          say "<< Done"
+        end
+      end
+
```
Comment (boatrite) [inline lib/subspace/commands/ansible.rb]: :dash: extra lines.

File: lib/subspace/commands/init.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      template "group_vars/template", "group_vars/#{env}"
+      template "host_vars/template", "host_vars/#{env}"
+      create_vars_file_for_env env
+      template "playbook.yml", "#{env}.yml"
+    end
+
+    puts """
+    1. Create a server.
+
+    2. Set your server's location:
```
Comment (boatrite) [inline lib/subspace/commands/init.rb]: None of these words are accurate anymore are they? Isn't this replaced with the ruby file?

File: template/provision.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+#   - playbook.yml
+
+# Step one: Configure your deployment environments.  We usually have a "dev" and a "production"
+# environment.  Sometimes we have a "staging" or qa.  You can do whatever you'd like.
+
+Subspace.configure do |config|
+  # Project level configuration
+
+  # Name of the project.  This will get used all over the place in path names, config file names, etc.
+  config.project_name = "<%= File.basename FileUtils.pwd =>"
```
Comment (boatrite) [inline template/provision.rb]: How does `"<%= File.basename FileUtils.pwd =>"` work / What does it do?

Comment (samsonasu) [inline template/provision.rb]: pwd = print working directory, this just gets the current directory name without the full path

**Pull request merged**